From 7e5ad5f5b9c7b0807337333c5622dfa61915d588 Mon Sep 17 00:00:00 2001
From: Song Bing <b06498@freescale.com>
Date: Fri, 15 Apr 2016 13:20:51 +0800
Subject: [PATCH] [application] Don't release texture when video playback.

Video will freeze a while when UI over the video. Don't release
texture can benefit a little.

Signed-off-by: Song Bing bing.song@nxp.com
---
 xbmc/Application.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index 7c0c9d5..bbef7ac 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -4525,9 +4525,10 @@ void CApplication::ProcessSlow()
   g_curlInterface.CheckIdle();
 
   if (!m_pPlayer->IsPlayingVideo())
+  {
     g_largeTextureManager.CleanupUnusedImages();
-
-  g_TextureManager.FreeUnusedTextures(5000);
+    g_TextureManager.FreeUnusedTextures(5000);
+  }
 
 #ifdef HAS_DVD_DRIVE
   // checks whats in the DVD drive and tries to autostart the content (xbox games, dvd, cdda, avi files...)
-- 
1.9.1

